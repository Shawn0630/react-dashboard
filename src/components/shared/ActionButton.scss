.actionButton {
    padding: 12px 6px !important;
    vertical-align: middle !important;
}