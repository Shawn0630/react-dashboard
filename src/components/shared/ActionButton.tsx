import * as React from "react";
import * as styles from "./ActionButton.scss";
import IconButton from "@material-ui/core/IconButton";
import Icon from "@material-ui/core/Icon";

interface ActionButtonProps<T> {
    data: T;
    title: string;
    icon: string;
    children?: any; //tslint:disable-line
    disabled?: boolean;
    style?: object;
    onClick(data: T): void;
}

function ActionButton<T>(props: ActionButtonProps<T>): JSX.Element { //tslint:disable-line
    function handleClick(event: React.SyntheticEvent<HTMLElement>): void {
        event.stopPropagation();
        props.onClick(props.data);
    }
    const disabled: boolean = props.disabled != null ? props.disabled : false;
    const style: object = props.style !== undefined ? props.style : {};

    return <IconButton disabled={disabled} className={styles.actionButton}
                       onClick={handleClick} title={props.title} style={style}>
        {props.children}
        <Icon>{props.icon}</Icon>
    </IconButton>;
}

export default ActionButton;
