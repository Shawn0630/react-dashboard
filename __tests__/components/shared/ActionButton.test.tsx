import * as React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import ActionButton from "~components/shared/ActionButton";
import IconButton from "@material-ui/core/IconButton";

describe("components/shared/ActionButton", () => {
    test("shoule match snapshot when enabled", () => {
        const onClick: (data: {}) => void = jest.fn();
        const title: string = "test title";
        const data: string = "test data";
        const disabled: boolean = false;

        const wrapper: ShallowWrapper = shallow(
            <ActionButton data={data} title={title} disabled={disabled} onClick={onClick} icon="mode_edit" />
        );
        expect(wrapper).toMatchSnapshot();

        const iconButton: ShallowWrapper = wrapper.find(IconButton).first();
        iconButton.simulate("click", {
            stopPropagation: jest.fn()
        });
        expect(onClick).toHaveBeenCalledWith(data);
    });
});
