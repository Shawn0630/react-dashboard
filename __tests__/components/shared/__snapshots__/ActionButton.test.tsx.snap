// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/ActionButton shoule match snapshot when enabled 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionButton
    data="test data"
    disabled={false}
    icon="mode_edit"
    onClick={[MockFunction]}
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        <WithStyles(Icon)>
          mode_edit
        </WithStyles(Icon)>,
      ],
      "className": "actionButton",
      "disabled": false,
      "onClick": [Function],
      "style": Object {},
      "title": "test title",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "mode_edit",
        },
        "ref": null,
        "rendered": "mode_edit",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          <WithStyles(Icon)>
            mode_edit
          </WithStyles(Icon)>,
        ],
        "className": "actionButton",
        "disabled": false,
        "onClick": [Function],
        "style": Object {},
        "title": "test title",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "mode_edit",
          },
          "ref": null,
          "rendered": "mode_edit",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
